### ISSUE TITLE
FK relation between model in public schema to model in shared schema - on delete gives doesn't exist error

### ISSUE THREAD

**Issue by m7salam** [django-tenants/django-tenants#592]
Is it possible to have a ForeignKey relation (eg: created_by field) from a model in a tenant schema (tenant app) to the User model which resides in the public schema (shared app) ?

when I try to delete the user for example in the public schema

it gives error because some of the tenant apps has FK relationship with it

how to fix this with django_tenants ???

**Comment by adnanmuttaleb** [comment]
It seems this is a major issue check:

https://github.com/django-tenants/django-tenants/issues/19

https://github.com/django-tenants/django-tenants/issues/68

I tried to place the `django.contrib.auth` in SHARED_APP, IN TENANT_APP and in both, but neither worked. I expect relationships with the `User` model as something essential, but it seems not supported without hacks.

**Comment by yujinyuz** [comment]
I'm not quite sure which issue I should comment this but I found a workaround for this one.

I've created a schema that is similar to one of the tenants but excluded it as clients. I named mine `public_fallback` and then added this in my `settings.py`

```python
PG_EXTRA_SEARCH_PATHS = ['public_fallback', ]
```

The idea is to take advantage of _search path_ used by postgres: https://www.postgresql.org/docs/current/ddl-schemas.html

So, for example you try to delete a user from the public tenant, Django will just attempt to find all other tables that is related to the user in the `public_fallback` schema and will attempt to delete it there.

TL;DR

1. Create a dummy schema e.g. `public_fallback` that contains all the tables similar to a regular tenant.
2. Add this to settings.py
```python
PG_EXTRA_SEARCH_PATHS = ['public_fallback', ]
```

**Comment by imtipu** [comment]
The only way I found that is working.
Using Django 4.2 and Python 3.11

You have to create an app as a tenant app, through admin or custom way. But, the name must be `public_fallback`. It will create all the tables same as all tenanta apps. Then delete will work. After creatjng this you should remove the tenant from tenant list in database. It will not delete the schema. Everytime you changes anything that's need migration after deployment you should resync this fallback schema.

**Comment by tomturner** [comment]
Closing as part of a backlog tidy-up — not because the report was wrong, but because it can no longer be acted on as it stands.

This has been inactive for some years and a great deal has changed underneath it: `clone_schema` has been replaced wholesale from upstream pg-clone-schema, psycopg3 is now the default driver, and the supported range is Django 4.2 / 5.2 / 6.0 on PostgreSQL 14–17. There's a fair chance this no longer reproduces.

If it still happens on **3.12.0**, please reopen with the django-tenants and Django versions, the PostgreSQL version, and the steps to reproduce — with that I can actually chase it down.

Thanks for reporting it.
